modules/FhirAPI/src/FhirAPI/FhirRestApiBuilder/Parts/Search/SearchStrategies/ServiceRequestSearch.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace FhirAPI\FhirRestApiBuilder\Parts\Search\SearchStrategies;
+
+
+class ServiceRequestSearch extends BaseSearch
+{
+    public $paramsToDB = array();
+    public $MAIN_TABLE = 'fhir_service_request';
+    public function search()
+    {
+
+        if(!empty($this->searchParams)){
+            foreach($this->searchParams as $index => $data){
+                if($index === "occurrence" || $index === "authored"){
+                    foreach($data as $pos => $record){
+                        $this->searchParams[$index][$pos]['value']=$this->fhirObj->convertToDateTime($record['value']);
+
+                    }
+                }
+            }
+        }
+
+        $this->paramHandler('_id','id');
+        $this->paramHandler('encounter','encounter');
+        $this->paramHandler('patient','patient');
+        $this->paramHandler('authored','authored_on');
```

Comment:
Did you check if the sort works correctly?
we need sort by authored DESC 
The request that the client-side will send is -  `ServiceRequest?encounter=1&patient=1&_sort=-authored`

Reply:
of course not testing is for QA :) 
I am in new install and I do not have data to test it . 
do you want me to defecate half a day to test the search ? 
if so I will add subtask

Reply:
I need to check what we are developing...
Yes, check it before, we need it to work correctly.

Reply:
every thing we need to do that does not take 5 min need to be sub task